haskell/src/Ligo/BaseDAO/Types.hs

```diff
@@ -811,19 +817,12 @@ instance CustomErrorHasDoc "bAD_ENTRYPOINT_PARAMETER" where
   customErrClass = ErrClassBadArgument
   customErrDocMdCause = "Value passed to the entrypoint is not valid"
 
-type instance ErrorArg "fAIL_DROP_PROPOSAL_NOT_OVER" = NoErrorArg
+type instance ErrorArg "dROP_PROPOSAL_CONDITION_NOT_MET" = NoErrorArg
 
-instance CustomErrorHasDoc "fAIL_DROP_PROPOSAL_NOT_OVER" where
+instance CustomErrorHasDoc "dROP_PROPOSAL_CONDITION_NOT_MET" where
   customErrClass = ErrClassActionException
   customErrDocMdCause =
-    "An error occurred when trying to drop a proposal due to the proposal's voting period is not over"
-
-type instance ErrorArg "fAIL_DROP_PROPOSAL_NOT_ACCEPTED" = NoErrorArg
-
-instance CustomErrorHasDoc "fAIL_DROP_PROPOSAL_NOT_ACCEPTED" where
-  customErrClass = ErrClassActionException
-  customErrDocMdCause =
-    "An error occurred when trying to drop a proposal due to the proposal is not an accepted proposal"
+    "Throw when trying to call `drop_proposal` when the sender is not proposer or guardian or proposal is not expired."
```

Comment:
> or proposal is not expired.

Not sure, but shouldn't this be "and proposal is not expired"?

Reply:
I think using `and` makes it sound like the proposals have to be expired. But it is not the case when the proposer or guardian make the call. They can drop the proposal regardless of whether or not it is expired.

Reply:
Yea, that is what I meant as well. The error happens only when caller is not proposer or guardian *and* the proposal is not expired. Right?

Using `or` there will make it as if it will throw error for proposal/guardian if the proposal is not expired. So right now it is like this `(not (proposer || guardian)) || (proposal is not expired)`, and when proposal is not expired this will always be true irrespective of the caller being proposer or guardian, which AFAIU is not correct.

Reply:
Make sense, I've updated this now.

Reply:
Just realize that the word `or` I use here, could be translated as `XOR` instead of `||`. Anyway, this could be an example case where natural language is quite ambiguous.